module/VuFind/src/VuFind/Controller/MyResearchController.php

```diff
@@ -1126,21 +1126,35 @@ public function deletelistAction()
     }
 
     /**
-     * Get a record driver object corresponding to an array returned by an ILS
-     * driver's getMyHolds / getMyTransactions method.
+     * Get record driver objects corresponding to an of record arrays returned by an
+     * ILS driver's methods such as getMyHolds / getMyTransactions.
      *
-     * @param array $current Record information
+     * @param array $records Record information
      *
-     * @return \VuFind\RecordDriver\AbstractBase
+     * @return array
```

Comment:
I think we could use the more specific typehint `\VuFind\RecordDriver\AbstractBase[]` here. (I realize that we currently use `array` all over the place, but this is one of the things that Psalm may encourage us to improve -- might as well start getting in the habit).

Reply:
Right, and I added type hints as well. Too bad you can't specify the type of array in return hint.

Reply:
Maybe eventually! PHP 8 at least makes aggregate typehints possible, so further improvements may come along eventually. For now, as long we capture things correctly in the PHPDoc, we can automatically fix the code if and when the language supports richer expression in the future.